collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java

```diff
@@ -88,7 +90,11 @@ public Outcome doInBackground(Long... instanceIdsToUpload) {
         Outcome outcome = new Outcome();
 
         InstanceServerUploader uploader = new InstanceServerUploader(httpInterface, webCredentialsUtils, settingsProvider.getUnprotectedSettings(), instancesRepository);
-        List<Instance> instancesToUpload = uploader.getInstancesFromIds(instanceIdsToUpload);
+        List<Instance> instancesToUpload = uploader
```

Comment:
It'd be really nice if we could just make this change in one place (probably the `InstanceSubmitter`). I'm guessing it's non-trivial to use that here?

Reply:
Yeah, InstanceSubmitter is not used here. I think we should file a separate issue for refactoring the whole InstanceUploaderTask as it shouldn't use the deprecated `AsyncTask` anymore and probably more code should be shared.